snapcraft/internal/sources/errors.py

```diff
@@ -72,8 +76,19 @@ def __init__(self, expected, calculated):
         super().__init__(expected=expected, calculated=calculated)
 
 
-class InvalidDebError(errors.SnapcraftError):
+class InvalidDebError(SnapcraftSourceError):
 
     fmt = ('The {deb_file} used does not contain valid data. '
            'Ensure a proper deb file is passed for .deb files '
            'as sources.')
+
+
+class SourceUpdateUnsupportedError(SnapcraftSourceError):
+
+    fmt = (
+        'Failed to update source: '
+        "{source!s} sources don't support updating"
```

Comment:
Can you add a `.` to end the sentence please?

Reply:
Done.